src/rpcblockchain.cpp

```diff
@@ -456,9 +465,13 @@ UniValue getblock(const UniValue& params, bool fHelp)
 
     uint256 hash(uint256S(strHash));
 
-    bool fVerbose = true;
-    if (params.size() > 1)
-        fVerbose = params[1].get_bool();
+    int verbosity = 1;
+    if (params.size() > 1) {
+        if(params[1].isNum())
+            verbosity = params[1].get_int();
+        else
+            verbosity = params[1].get_bool() ? 1 : 0;
+    }
```

Comment:
Add a check to make sure the verbosity parameter is valid.  Right now, it is possible to pass in -1 or 3 and no error is thrown, when one should be.